lib/msal-browser/src/interaction_client/StandardInteractionClient.ts

```diff
@@ -270,23 +270,35 @@ export abstract class StandardInteractionClient extends BaseInteractionClient {
             validatedRequest.account = account;
         }
 
-        // Check for ADAL SSO
+        // Check for ADAL/MSAL v1 SSO
         if (StringUtils.isEmpty(validatedRequest.loginHint)) {
-            // Only check for adal token if no SSO params are being used
+            // Only check for adal/msal token if no SSO params are being used
             const adalIdTokenString = this.browserStorage.getTemporaryCache(PersistentCacheKeys.ADAL_ID_TOKEN);
             if (adalIdTokenString) {
-                const adalIdToken = new IdToken(adalIdTokenString, this.browserCrypto);
+                this.logger.verbose("Cached ADAL id token retrieved.");
+            }
+
+            // Check for cached MSAL v1 id token
+            const msalIdTokenString = this.browserStorage.getTemporaryCache(PersistentCacheKeys.ID_TOKEN, true);
+            if (msalIdTokenString) {
+                this.logger.verbose("Cached MSAL.js v1 id token retrieved");
+            }
+
+            const cachedIdTokenString = msalIdTokenString || adalIdTokenString;
+            if (cachedIdTokenString) {
+                const cachedIdToken = new IdToken(cachedIdTokenString, this.browserCrypto);
                 this.browserStorage.removeItem(PersistentCacheKeys.ADAL_ID_TOKEN);
-                if (adalIdToken.claims && adalIdToken.claims.preferred_username) {
-                    this.logger.verbose("No SSO params used and ADAL token retrieved, setting ADAL preferred_username as loginHint");
-                    validatedRequest.loginHint = adalIdToken.claims.preferred_username;
+                this.browserStorage.removeItem(this.browserStorage.generateCacheKey(PersistentCacheKeys.ID_TOKEN));
```

Comment:
Is the result from generateCacheKey a constant in these flows?  Do we need to make this call or can we define a constant?

Reply:
It is more complicated, unfortunately. The other browser storage APIs include this logic baked in (configurable via a flag), however, the `removeItem` API does not do that. Probably a fix we should make, but can be done separately, as it would be a larger PR.